### PR TITLE
Adding Labram

### PR DESCRIPTION
**Pull request by bruAristimunha**

### CODE REVIEW

File: braindecode/models/modules.py [hunk 1]
```diff
@@ -356,3 +359,693 @@ def forward(self, x):
...
+    def __init__(self, n_times=2000, patch_size=200, n_chans=1, embed_dim=200):
+        super().__init__()
+
+        self.n_times = n_times
+        self.patch_size = patch_size
+        self.n_patchs = n_times // patch_size
+        self.embed_dim = embed_dim
+        self.n_chans = n_chans
+
+        self.patcher = nn.Conv2d(
```
[the remark this answers is not in the document]
Comment (bruAristimunha) [reply braindecode/models/modules.py:1002]: Unfold is no a good option...

File: braindecode/models/modules.py [hunk 2]
```diff
@@ -356,3 +359,693 @@ def forward(self, x):
...
+
+    def forward(self, x):
+        return drop_path(x, self.drop_prob, self.training)
+
+    # Utility function to print DropPath module
+    def extra_repr(self) -> str:
+        return f"p={self.drop_prob}"
+
+
+class Attention(nn.Module):
```
[the remark this answers is not in the document]
Comment (bruAristimunha) [reply braindecode/models/modules.py:619]: moved

**Comment by bruAristimunha** [comment]
Hi @agramfort!

I'm adding the [Labram](https://openreview.net/forum?id=QzTpTRVtrP) model, ICLR spotlight 2024. It says it scales BIOT results with Temple University. The model itself wasn't that clean, so I spent a lot of time cleaning up the code, commenting, and interacting with the author to answer some questions (https://github.com/935963004/LaBraM)

The model has a lot of inspiration from another computer vision model called BITv2, so many decisions come from elsewhere. BiTv2 is a 2022 visual transformer with attention windows.

The model scales well in the article, but I didn't conduct a more exhaustive self-supervised test to test it. I hope to do this when I finish the integration with another model that studies better self-supervised EEG (https://github.com/Navidfoumani/EEG2Rep/issues/1).

Thanks in advance for your review!

File: braindecode/models/modules.py [hunk 3]
```diff
@@ -356,3 +358,82 @@ def forward(self, x):
         return F.conv2d(
             x, weight=combined_weight, bias=bias, stride=(1, 1)
         )
+
+
+class MLP(nn.Module):
```
Comment (agramfort) [inline braindecode/models/modules.py:363]: I would use an MLP like this one https://github.com/probabilists/zuko/blob/master/zuko/nn.py#L122
here it's just one hidden layer.

**Review by agramfort — COMMENTED** [review]
could you add a test like for other networks?

File: braindecode/models/modules.py (hunk 3 above)
Comment (bruAristimunha) [reply braindecode/models/modules.py:363]: Thank you for this suggestion. I liked the Zuko library!

File: braindecode/models/functions.py [hunk 4]
```diff
@@ -45,3 +59,53 @@ def transpose_time_to_spat(x):
         tensor in which last and first dimensions are swapped
     """
     return x.permute(0, 3, 2, 1)
+
+
+def drop_path(x,
```
Comment (bruAristimunha) [inline braindecode/models/functions.py:64]: how to test this?

File: braindecode/models/labram.py
Comment (bruAristimunha) [inline braindecode/models/labram.py:1]: Test ideas: 
- parameter size, 
- functional with neural tokenizer and with neural decoder. 
- "Embedding" test to check the batch size; the model seems to work only with large batch.
- BioT correctness test?
- All the forward options.

- With pos embedding and without positional embedding; 

- Test for my seg patch function (is the output shape enough?)
- _PatchEmbed (some test)...

File: braindecode/models/modules.py [hunk 5]
```diff
@@ -356,3 +358,113 @@ def forward(self, x):
...
+        layers = layers[:-2]
+        layers.append(nn.Dropout(p=drop))
+
+        # Cleaning if we are not using the normalization layer
+        layers = filter(lambda layer: layer is not None, layers)
+
+        super().__init__(*layers)
+
+
+class DropPath(nn.Module):
```
Comment (bruAristimunha) [inline braindecode/models/modules.py:436]: how to test this part?

Comment (agramfort) [reply braindecode/models/modules.py:436]: can you at least test that the forward pass works on some random data with the right shape? I would basically just check success of forward and shape of outputs.

**Review by agramfort — APPROVED** [review]

**Pull request merged**